python/pyspark/streaming/kafka010.py

```diff
@@ -0,0 +1,370 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.rdd import RDD
+from pyspark.serializers import AutoBatchedSerializer, PickleSerializer
+from pyspark.streaming import DStream
+from pyspark.streaming.kafka import KafkaDStream, KafkaRDD, OffsetRange
+
+__all__ = ['Assign', 'KafkaConsumerRecord', 'KafkaUtils', 'PreferBrokers', 'PreferConsistent',
+           'PreferFixed', 'Subscribe', 'SubscribePattern', 'TopicPartition', 'utf8_decoder']
+
+
+def utf8_decoder(s):
+    """ Decode the unicode as UTF-8 """
+    if s is None:
+        return None
+    return s.decode('utf-8')
+
+
+class KafkaUtils(object):
+
+    @staticmethod
+    def createDirectStream(ssc, locationStrategy, consumerStrategy,
+                           keyDecoder=utf8_decoder, valueDecoder=utf8_decoder):
+        """
+        .. note:: Experimental
+
+        Create an input stream that directly pulls messages from Kafka 0.10 brokers with different
+        location strategy and consumer strategy.
+
+        This does not use Zookeeper to store offsets. The consumed offsets are tracked
```

Comment:
I don't think this verbiage is as relevant to Kafka 0.10, since the kafka managed api for offsets uses kafka topics, not zookeeper.